python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -322,14 +324,42 @@ async def slew_dome_to(self, az, check=None):
         # edit here and let it as is.
         _check.atmcs = False
 
-        task_list = [
-            asyncio.create_task(
-                self.wait_for_atdome_inposition(timeout=self.long_long_timeout)
-            ),
-            asyncio.create_task(self.monitor_position(check=_check)),
-        ]
+        monitor_position_task = asyncio.create_task(self.monitor_position(check=_check))
+
+        try:
+            for i in range(self._dome_slew_max_iter):
+
+                self.log.debug(
+                    f"[{i+1/self._dome_slew_max_iter}] Slewing dome to {az}..."
+                )
+
+                await self._handle_in_position(
+                    in_position_event=self.rem.atdome.evt_azimuthInPosition,
+                    timeout=self.long_long_timeout,
+                    settle_time=self.tel_settle_time,
+                    component_name="ATDome",
+                )
+                dome_azimuth_in_position = (
+                    await self.rem.atdome.evt_azimuthInPosition.aget(
+                        timeout=self.fast_timeout
+                    )
+                )
+
+                if dome_azimuth_in_position.inPosition:
+                    self.log.debug("Dome in position.")
+                    break
+                else:
+                    self.log.debug(
+                        "Dome not in position, probably overshoot. Repositioning..."
```

Comment:
Is there a way to determine whether the dome overshoots or not? The ATDome seems to emit an event when it is in position so I guess it also reports its position. Maybe something to look into in the future.

Reply:
Also, this message, which indicates that the dome will be repositioned, will be logged even if the maximum number of iterations has been reached. Maybe consider emitting an event in such a case? That may help to analyze how often this happens so it can be fixed?